asyncapi-cli/src/main/java/io/ballerina/asyncapi/cmd/AsyncApiCmd.java

```diff
@@ -124,6 +220,146 @@ public void execute() {
         }
     }
 
+    private void giveWarningsForInvalidSpecGenOptions() {
+        if (licenseFilePath != null) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, LICENSE_FLAG, SPEC));
+        }
+        if (includeTestFiles) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, TEST_FLAG, SPEC));
+        }
+    }
+
+    private void giveWarningsForInvalidClientGenOptions() {
+        if (generatedFileType) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, JSON_FLAG, CLIENT));
+        }
+        if (service != null) {
+            outStream.println(String.format(INVALID_OPTION_WARNING, SERVICE_FLAG, CLIENT));
+        }
+    }
+
+    private void verifyValidInputsForHttp() {
+        if (licenseFilePath != null) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, LICENSE_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+        if (service != null) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, SERVICE_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+        if (includeTestFiles) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, TEST_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+        if (generatedFileType) {
+            outStream.println(String.format(INVALID_OPTION_ERROR_HTTP, JSON_FLAG));
+            exitError(this.exitWhenFinish);
+        }
+    }
+
+    private void ballerinaToAsyncApiWs(String fileName) {
+        List<AsyncApiConverterDiagnostic> errors = new ArrayList<>();
+        final File balFile = new File(fileName);
+        Path balFilePath = null;
+        try {
+            balFilePath = Paths.get(balFile.getCanonicalPath());
+        } catch (IOException e) {
+            DiagnosticMessages message = DiagnosticMessages.AAS_CONVERTOR_102;
+            ExceptionDiagnostic error = new ExceptionDiagnostic(message.getCode(),
+                    message.getDescription(), null, e.getLocalizedMessage());
+            errors.add(error);
+        }
+        getTargetOutputPathWs();
+        // Check service name it is mandatory
+        BallerinaToAsyncApiGenerator asyncApiConverter = new BallerinaToAsyncApiGenerator();
+        asyncApiConverter.generateAsyncAPIDefinitionsAllService(balFilePath, targetOutputPath, service,
+                generatedFileType);
+
+        errors.addAll(asyncApiConverter.getErrors());
```

Comment:
According to this my assumption in this comment [1] is invalid. But, I am wondering why can't we return these errors like this ?
```java
List<AsyncApiConverterDiagnostic> conversionErrors = asyncApiConverter.generateAsyncAPIDefinitionsAllService(balFilePath, targetOutputPath, service,
                generatedFileType);
```

IMO the asyn-api generator should be stateless. But, if there is a requirement it to be state-full then we could have provided these parameters in the constructor itself and invoke the `generateAsyncAPIDefinitionsAllService` method and then retrieve errors from the converter. 

[1] - https://github.com/ballerina-platform/asyncapi-tools/pull/119/files#r1656424939

Reply:
+1, I will refactor this.